rosbag2_py/test/test_compression.py

```diff
@@ -0,0 +1,96 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+from common import get_rosbag_options
+
+import pytest
+
+from rclpy.serialization import serialize_message
+from rosbag2_py import (
+    compression_mode_from_string,
+    compression_mode_to_string,
+    CompressionMode,
+    CompressionOptions,
+    SequentialCompressionWriter,
+    TopicMetadata,
+)
+from rosbag2_test_common import TESTED_STORAGE_IDS
+
+from std_msgs.msg import String
+
+
+def test_compression_mode_from_string():
+    """Checks that we can cast a string to a compression mode."""
+    assert CompressionMode.NONE == compression_mode_from_string('NONE')
+    assert CompressionMode.MESSAGE == compression_mode_from_string('MESSAGE')
+    assert CompressionMode.FILE == compression_mode_from_string('FILE')
+
+
+def test_compression_mode_to_string():
+    """Checks that we can cast a compression mode to a string."""
+    assert 'NONE' == compression_mode_to_string(CompressionMode.NONE)
+    assert 'MESSAGE' == compression_mode_to_string(CompressionMode.MESSAGE)
+    assert 'FILE' == compression_mode_to_string(CompressionMode.FILE)
+
+
+def test_compression_options():
+    """Checks that we can construct a CompressionOptions class."""
+    compression_options = CompressionOptions(
+        compression_format='zstd',
+        compression_mode=CompressionMode.MESSAGE,
+        compression_queue_size=0,
+        compression_threads=8)
+    assert compression_options is not None
+    assert compression_options.compression_format == 'zstd'
+    assert compression_options.compression_mode == CompressionMode.MESSAGE
+    assert compression_options.compression_queue_size == 0
+    assert compression_options.compression_threads == 8
+
+
+@pytest.mark.parametrize('storage_id', TESTED_STORAGE_IDS)
+def test_sequential_compression_writer(tmp_path, storage_id):
+    """Checks that we can do a compressed write and read."""
+    bag_path = os.path.join(tmp_path, 'tmp_sequential_compressed_write_test')
+
+    storage_options, converter_options = get_rosbag_options(
+        path=bag_path,
+        storage_id=storage_id)
+
+    compression_options = CompressionOptions(
+        compression_format='zstd',
+        compression_mode=CompressionMode.MESSAGE,
+        compression_queue_size=0,
+        compression_threads=1)
+
+    writer = SequentialCompressionWriter(compression_options)
+    writer.open(storage_options, converter_options)
+
+    topic_name = '/chatter'
+    topic_metadata = TopicMetadata(
+        name=topic_name,
+        type='std_msgs/msg/String',
+        serialization_format='cdr')
+    writer.create_topic(topic_metadata)
+
+    for i in range(10):
+        msg = String()
+        msg.data = f'Hello, world! {str(i)}'
+        time_stamp = i * 100
+
+        writer.write(topic_name, serialize_message(msg), time_stamp)
+
+    # close bag and create new storage instance
+    del writer
```

Comment:
Need to verify that we have successfully written rosbag2.   
Could you please add similar checks as in the ?https://github.com/ros2/rosbag2/blob/744cdf0420189b4435b9c8b2839899b1b80d23f5/rosbag2_py/test/test_sequential_writer.py#L70-L90

Reply:
Done.